File: movemark-android/app/src/main/java/com/surensureshkumar/movemark/features/auth/AuthProofVaultStrip.kt
package com.surensureshkumar.movemark.features.auth

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.surensureshkumar.movemark.core.design.MMColors

private val stripRaised = Color(0xFF1A1C1B)
private val trustItems = listOf(
    "Private proof vault",
    "Move-in report included",
    "Upgrade later for move-out and disputes",
)

/**
 * Compact auth proof context — vault strip or trust rows. No dashboard card.
 */
@Composable
fun AuthProofVaultStrip(
    mode: AuthMode,
    modifier: Modifier = Modifier,
) {
    val shape = RoundedCornerShape(14.dp)
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(shape)
            .background(stripRaised)
            .border(0.75.dp, MMColors.CardStroke.copy(alpha = 0.55f), shape)
            .padding(horizontal = 12.dp, vertical = 11.dp),
    ) {
        if (mode == AuthMode.SignIn) {
            Row(Modifier.fillMaxWidth()) {
                Icon(
                    Icons.Filled.CheckCircle,
                    contentDescription = null,
                    tint = MMColors.Primary.copy(alpha = 0.9f),
                    modifier = Modifier
                        .size(15.dp)
                        .padding(top = 1.dp),
                )
                Spacer(Modifier.width(10.dp))
                Column {
                    Text(
                        "Proof vault ready",
                        fontSize = 14.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = MMColors.TextPrimary,
                        maxLines = 1,
                    )
                    Text(
                        "Pick up where your room proof left off.",
                        fontSize = 12.5.sp,
                        fontWeight = FontWeight.Medium,
                        color = MMColors.TextSecondary,
                        modifier = Modifier.padding(top = 3.dp),
                        maxLines = 2,
                    )
                    Text(
                        "Move-in proof · Reports when needed",
                        fontSize = 11.sp,
                        fontWeight = FontWeight.Medium,
                        color = MMColors.TextMuted,
                        modifier = Modifier.padding(top = 4.dp),
                        maxLines = 1,
                    )
                }
            }
        } else {
            trustItems.forEachIndexed { index, label ->
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(top = if (index == 0) 0.dp else 8.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Icon(
                        Icons.Filled.CheckCircle,
                        contentDescription = null,
                        tint = MMColors.Primary.copy(alpha = 0.88f),
                        modifier = Modifier.size(13.dp),
                    )
                    Spacer(Modifier.width(8.dp))
                    Text(
                        label,
                        fontSize = 12.5.sp,
                        fontWeight = FontWeight.Medium,
                        color = MMColors.TextSecondary,
                        maxLines = 2,
                    )
                }
            }
        }
    }
}

File: movemork/Features/Auth/AuthProofVaultStrip.swift
//
//  AuthProofVaultStrip.swift
//  movemork
//
//  Compact auth proof context — vault strip or trust rows. No dashboard card.
//

import SwiftUI

struct AuthProofVaultStrip: View {
    let mode: AuthContainerView.Mode

    private let stripRadius: CGFloat = 14
    private let trustItems = [
        "Private proof vault",
        "Move-in report included",
        "Upgrade later for move-out and disputes",
    ]

    var body: some View {
        Group {
            if mode == .signIn {
                signInStrip
            } else {
                signUpTrustRows
            }
        }
        .id("auth-strip-\(mode == .signIn ? "in" : "up")")
    }

    private var signInStrip: some View {
        HStack(alignment: .top, spacing: 10) {
            Image(systemName: "checkmark.circle.fill")
                .font(MoveMarkTheme.Typography.subheadlineMedium)
                .foregroundStyle(MoveMarkTheme.Colors.primary.opacity(0.9))
                .padding(.top, 1)

            VStack(alignment: .leading, spacing: 3) {
                Text("Proof vault ready")
                    .font(MoveMarkTheme.Typography.detailEmphasis)
                    .foregroundStyle(MoveMarkTheme.Colors.textPrimary)
                    .lineLimit(1)

                Text("Pick up where your room proof left off.")
                    .font(MoveMarkTheme.Typography.caption)
                    .foregroundStyle(MoveMarkTheme.Colors.textSecondary)
                    .lineLimit(2)
                    .fixedSize(horizontal: false, vertical: true)

                Text("Move-in proof · Reports when needed")
                    .font(MoveMarkTheme.Typography.tinyMedium)
                    .foregroundStyle(MoveMarkTheme.Colors.textMuted)
                    .lineLimit(1)
                    .padding(.top, 1)
            }

            Spacer(minLength: 0)
        }
        .padding(.horizontal, 12)
        .padding(.vertical, 11)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(stripBackground)
    }

    private var signUpTrustRows: some View {
        VStack(alignment: .leading, spacing: 8) {
            ForEach(trustItems, id: \.self) { item in
                trustRow(item)
            }
        }
        .padding(.horizontal, 12)
        .padding(.vertical, 11)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(stripBackground)
    }

    private func trustRow(_ label: String) -> some View {
        HStack(alignment: .center, spacing: 8) {
            Image(systemName: "checkmark.circle.fill")
                .font(MoveMarkTheme.Typography.footnoteEmphasis)
                .foregroundStyle(MoveMarkTheme.Colors.primary.opacity(0.88))

            Text(label)
                .font(MoveMarkTheme.Typography.caption)
                .foregroundStyle(MoveMarkTheme.Colors.textSecondary)
                .lineLimit(2)
                .fixedSize(horizontal: false, vertical: true)
        }
    }

    private var stripBackground: some View {
        RoundedRectangle(cornerRadius: stripRadius, style: .continuous)
            .fill(MoveMarkTheme.Colors.evidenceCardRaised.opacity(0.92))
            .overlay(
                RoundedRectangle(cornerRadius: stripRadius, style: .continuous)
                    .stroke(MoveMarkTheme.Colors.cardStroke.opacity(0.55), lineWidth: 0.75)
            )
    }
}
